fsm_codegen/src/codegen.rs

```diff
@@ -19,27 +19,19 @@ pub fn build_state_store(fsm: &FsmDescription) -> quote::Tokens {
     let mut n = quote::Tokens::new();
     for state in &fsm.get_all_states() {
         let field_name = FsmDescription::to_state_field_name(&state);
-        f.append(quote! { #field_name: #state,  }.as_str());
-        n.append(quote! { #field_name: #state::new_state(context), }.as_str());
+        f.append(quote! { #field_name: FsmArc<#state>,  }.as_str());
+        n.append(quote! { #field_name: std::sync::Arc::new(tokio::sync::RwLock::new(#state::new_state(context))), }.as_str());
```

Comment:
Why no include?

Reply:
Because it's generated code that gets injected into a user rust file and I don't want to influence the global scope with my includes.